### DISCUSSION TITLE
Two listboxes scrolling in synch

### DISCUSSION THREAD

**Discussion by renrutsirhc** [AvaloniaUI/Avalonia#6037] [Q&A] [answered]
Hi all,

Can anyone suggest an approach for having two listboxes scroll in synch (the listbox items in each will share the same height!). I've found various articles about doing this in WPF but the properties/events needed don't seem to exist in Avalonia as far as I can tell. Any thoughts?

Chris

**Comment by renrutsirhc** [comment, marked as answer]
OK - so while I didn't work out how to do this accessing the scrollviewer inside the Listbox, I found a workaround. I created a new version of ListBox and removed all the scrolling code. I could then use the following UI structure:
```
<ScrollViewer>
<Grid>
<NonScrollingListBox/>
<NonScrollingListBox/>
</Grid>
</ScrollViewer>
```
The scrollviewer outside the grid then scrolls both nonscrollinglistboxes together.

If anyone has any insight on accessing the properties of the scrollviewer inside a normal listbox, I'd greatly appreciate it for my own learning!

**Reply by maxkatz6** [reply]
This approach is correct one, with which you don't need to care about inner scrolls synchronization.
But instead of creating your own ListBox you can just disable scrolling on existing one:
```xaml
<ListBox ScrollViewer.VersticalScrollVisibility="Disabled" />
```
(Property name might be different, wrote from memory)

**Reply by renrutsirhc** [reply]
Thanks @maxkatz6 - I tried that initially but was getting some weird behavior.

The Listboxes were not scrolling to their full extent using the scrollbar. There was also the issue that the mousewheel events were still being captured by the listboxes and they were therefore being scrolled independently when using the mousewheel. Perhaps there is another solution to that?

**Reply by maxkatz6** [reply]
Oh, I see what you mean. It seems this approach is broken because of virtualization. 
If you set ```<ListBox VirtualizationMode="None" ... />``` as well, then it will work. But obviously it's not acceptable if you have huge lists. Though with your approach (creating custom listboxes without scroll) will have the same problem with missing virtualization.
And now I am thinking that not sure if it's possible to keep it working in any way. Most likely ItemsRepeater will allow it, but you will need to implement items selection manually.

**Reply by renrutsirhc** [reply]
Thanks for the suggestion Max - the VirtualizationMode trick solved it for normal listboxes (the lists I am working with are not huge so performance is not negatively impacted. This is what I will use for this app.

Also, it did work fine with the custom listbox without the scrolling stuff even with virtualisation turned on.

**Reply by maxkatz6** [reply]
> Also, it did work fine with the custom listbox without the scrolling stuff even with virtualisation turned on.

Most likely it just didn't work.
